packages/client/src/builders/search/productHighlightingBuilder.ts

```diff
@@ -0,0 +1,50 @@
+import { HighlightSettings2ProductProductHighlightPropsHighlightSettings2Limits, HighlightSettings2ProductProductHighlightPropsHighlightSettings2ResponseShape, ProductHighlightProps, ProductSearchSettingsHighlightSettings } from '../../models/data-contracts';
+
+export class ProductHighlightingBuilder {
```

Comment:
Could we have had a base builder to unify some of the common code? Like the enabled and shape could have been part of a base builder

Reply:
The shape and limit are its own unique type for either product or content. I could not find a good way to consolidate them. 😶